tests/unit/charms/lte_core_interface/v0/test_lte_core_interface_provider.py

```diff
@@ -63,28 +65,30 @@ def test_given_mme_address_is_not_valid_when_update_relation_data_then_value_err
     ):
         self.harness.set_leader(is_leader=True)
         mme_ipv4_address = "invalid ipv4 address"
-        relation_id = self.harness.add_relation(
+        self.harness.add_relation(
             relation_name=self.relation_name, remote_app=self.remote_app_name
         )
-        self.harness.update_relation_data(
-            relation_id=relation_id,
-            app_or_unit=self.remote_app_name,
-            key_values={"mme_ipv4_address": mme_ipv4_address},
-        )
         with pytest.raises(AddressValueError) as e:
             self.harness.charm.lte_core_provider.set_lte_core_information(
                 mme_ipv4_address=mme_ipv4_address
             )
 
         self.assertEqual(str(e.value), "Invalid MME IPv4 address.")
 
-    def test_given_relation_not_created_when_set_lte_core_information_then_runtime_error_is_raised(  # noqa: E501
-        self,
+    @patch(
+        "charms.lte_core_interface.v0.lte_core_interface.LTECoreProvides.set_lte_core_information"
+    )
+    def test_given_mme_address_is_not_valid_when_update_relation_data_then_status_is_blocked(  # noqa: E501
+        self, patch_set_lte_core_information
     ):
         self.harness.set_leader(is_leader=True)
-        mme_ipv4_address = "0.0.0.0"
+        patch_set_lte_core_information.side_effect = AddressValueError("Invalid MME IPv4 address.")
+        self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=self.remote_app_name
+        )
+        relation_joined_event = Mock()
+        self.harness.charm._on_lte_core_relation_joined(relation_joined_event)
 
-        with pytest.raises(RuntimeError):
-            self.harness.charm.lte_core_provider.set_lte_core_information(
-                mme_ipv4_address=mme_ipv4_address
-            )
+        self.assertEqual(
+            self.harness.charm.unit.status, BlockedStatus("Invalid MME IPv4 address.")
```

Comment:
Our test should validate the behavior of the lib, not the dummy charm, Here the lib raises so that's what we should validate.

Reply:
We are "imposing" an implementation of `_on_lte_core_relation_joined` in the initial docstring and in the `README`, don't we consider that as "part" of the lib? If we don't, we cannot ensure the same behaviour. The user could not even call the `self.lte_core_provider.set_lte_core_information(mme_ipv4_address="0.0.0.0")` and the lib won't work at all. I don't know what I'm missing here. I'd like to get it before I make the change.

Reply:
My point here is that the dummy charm has some logic in it and that's what this test is validating. This logic is something like when set_lte_core_information raises a `AddressValueError`, go to blocked status. We don't really care about this logic ultimately, what we care about is the lib itself. What we should validate is when set_lte_core_information is called with a bad IP address, it raises `AddressValueError`